app/javascript/components/cloud-volume-form/index.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './cloud-volume-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const CloudVolumeForm = ({ recordId }) => {
+  const [{ fields, initialValues, isLoading }, setState] = useState({ fields: [], isLoading: !!recordId });
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+  const promise = useMemo(() =>
+    API.get('/api/providers?expand=resources&attributes=id,name,supports_block_storage&filter[]=supports_block_storage=true'), []);
+
+  const loadSchema = (value, appendState = {}) => {
+    console.log(value);
+    setState(state => ({
+      ...state,
+      ...appendState,
+    }));
+  };
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/providers/${recordId}`).then((initialValues) => {
+        promise.then(loadSchema(initialValues.type, { initialValues, isLoading: false }));
+      });
+    }
+  }, []);
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+
+    const request = recordId ? API.patch(``, values) : API.post('', values);
+    request.then(() => {
+      const message = sprintf(
+        recordId
+          ? __('Modification of Cloud Volume "%s" has been successfully queued.') //Add Backup as third option
+          : __('Add of Cloud Volume "%s" has been successfully queued.'),
+        values.name,
+      );
+      miqRedirectBack(message, undefined, '/cloud_volume/show_list');
+    }).catch(miqSparkleOff);
+  };
+
+  const onCancel = () => {
+    const message = sprintf(
+      recordId
+        ? __('Edit of Cloud Volume "%s" was canceled by the user.') //Add Backup as third option
```

Comment:
backup?

Reply:
I noticed this form has some additional options and assumed they needed to be included as well
<img width="1403" alt="Screen Shot 2020-10-09 at 9 20 12 AM" src="https://user-images.githubusercontent.com/64800041/95588057-eacd2700-0a10-11eb-9324-6824c977bffe.png">

Reply:
Nope, those are different things, not related to your form.